db/migrate/20241127072350_remap_fa5_icon_names_to_fa6.rb

```diff
@@ -0,0 +1,769 @@
+    "tombstone-alt" => "tombstone-blank",
+    "tools" => "screwdriver-wrench",
+    "torah" => "scroll-torah",
+    "tram" => "train-tram",
+    "transgender-alt" => "transgender",
+    "trash-alt" => "trash-can",
+    "trash-restore" => "trash-arrow-up",
+    "trash-restore-alt" => "trash-can-arrow-up",
+    "trash-undo-alt" => "trash-can-undo",
+    "tree-alt" => "tree-deciduous",
+    "triangle-music" => "triangle-instrument",
+    "trophy-alt" => "trophy-star",
+    "truck-couch" => "truck-ramp-couch",
+    "truck-loading" => "truck-ramp-box",
+    "tshirt" => "shirt",
+    "tv-alt" => "tv",
+    "undo" => "arrow-rotate-left",
+    "undo-alt" => "rotate-left",
+    "university" => "building-columns",
+    "unlink" => "link-slash",
+    "unlock-alt" => "unlock-keyhole",
+    "usd-circle" => "circle-dollar",
+    "usd-square" => "square-dollar",
+    "user-alt" => "user-large",
+    "user-alt-slash" => "user-large-slash",
+    "user-chart" => "chart-user",
+    "user-circle" => "circle-user",
+    "user-cog" => "user-gear",
+    "user-edit" => "user-pen",
+    "user-friends" => "user-group",
+    "user-hard-hat" => "user-helmet-safety",
+    "user-md" => "user-doctor",
+    "user-md-chat" => "user-doctor-message",
+    "user-times" => "user-xmark",
+    "users-class" => "screen-users",
+    "users-cog" => "users-gear",
+    "users-crown" => "user-group-crown",
+    "utensil-fork" => "fork",
+    "utensil-knife" => "knife",
+    "utensil-spoon" => "spoon",
+    "utensils-alt" => "fork-knife",
+    "vhs" => "cassette-vhs",
+    "volleyball-ball" => "volleyball",
+    "volume-down" => "volume-low",
+    "volume-mute" => "volume-xmark",
+    "volume-up" => "volume-high",
+    "vote-nay" => "xmark-to-slot",
+    "vote-yea" => "check-to-slot",
+    "walking" => "person-walking",
+    "warehouse-alt" => "warehouse-full",
+    "washer" => "washing-machine",
+    "water-lower" => "water-arrow-down",
+    "water-rise" => "water-arrow-up",
+    "waveform-path" => "waveform-lines",
+    "webcam" => "camera-web",
+    "webcam-slash" => "camera-web-slash",
+    "weight" => "weight-scale",
+    "wifi-1" => "wifi-weak",
+    "wifi-2" => "wifi-fair",
+    "window-alt" => "window-flip",
+    "window-close" => "rectangle-xmark",
+    "wine-glass-alt" => "wine-glass-empty",
+  }
+  SVG_ICON_SUBSET_SETTING_NAME = "svg_icon_subset"
+
+  def up
+    migrate_svg_icon_subset_site_setting
+    migrate_icon_names
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
+  end
+
+  def migrate_svg_icon_subset_site_setting
+    # name is unique across site settings
+    setting =
+      execute(
+        "SELECT value FROM site_settings WHERE name = '#{SVG_ICON_SUBSET_SETTING_NAME}'",
+      ).first
+    return if !setting
+
+    original_setting_value = setting["value"]
+
+    new_setting_value =
+      original_setting_value
+        .split("|")
+        .map do |icon_name|
+          found_icon_name =
+            case icon_name
```

Comment:
This mimics the same logic that's in `migrate_icon_names` and follows the same remapping logic we have here: https://github.com/discourse/discourse/blob/fad0d2ec62187ce8ee749fd03fd7006a86f5c20f/lib/svg_sprite.rb#L522-L523.

Reply:
Doing this remapping of a `list` SiteSetting in SQL will be quite unreadable so decided to pull it out and do it in ruby.